### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by vimalsuresh**
## What does this pull request do?
Add circle ci config
Dockerfile for docker image
kubernetes pod definition
Helm definition for kube cluster
report builder
ruby-oci8 gem for running inside docker

## Why make these changes?

To enable running feature tests in pipeline

## Checklist

https://dsdmoj.atlassian.net/browse/TA-2371

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,5 +9,14 @@ group :test do
   gem 'rspec', '~> 3.10.0'
   gem 'selenium-webdriver', '~> 3.142.7'
   gem 'site_prism', '~> 3.7.3'
+  gem 'execjs'
+  gem 'date'
+  gem 'sequel'
+  gem 'report_builder', '~> 1.9'
   gem "timecop", "~> 0.9.8"
+  gem "parallel_tests"
```
Comment (solomonahmed) [inline Gemfile:17]: Add README.md detailing in brief info on gems here. 
Remove unnecessary gems as discussed.

File: charts/cwa-tests/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
```
Comment (solomonahmed) [inline charts/cwa-tests/Chart.yaml:1]: Add either README.md or put info in Wiki page providing details on contents and use of yaml file here.

File: charts/cwa-tests/Values.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+global:
```
Comment (solomonahmed) [inline charts/cwa-tests/Values.yaml:1]: Add either README.md or put info in Wiki page providing details on contents and use of yaml file here.

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,12 +1,58 @@
 version: 2.1
 
 orbs:
-  browser-tools: circleci/browser-tools@1.4.0
+  aws-cli: circleci/aws-cli@4.0.0
```
Comment (solomonahmed) [inline .circleci/config.yml:4]: Add either README.md or put info in Wiki page providing details on contents and use of yaml file here.

File: charts/cwa-tests/templates/pod.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+apiVersion: v1
```
Comment (solomonahmed) [inline charts/cwa-tests/templates/pod.yml:1]: Add either README.md or put info in Wiki page providing details on contents and use of yaml file here.

File: features/provider_bulk_load.feature [hunk 6]
```diff
@@ -10,12 +10,10 @@ Feature: Provider Bulk load
     Given a test firm user is logged in CWA
     When user bulk loads "valid.csv" for the test firm
     Then user should see the outcome results page
-    And the following summary for the submission:
-      | Total Outcomes | Successful Outcomes | Problem Outcomes | Duplicate Outcomes | Invalid Outcomes/NMS |
-      | 3              | 1                   | 1                | 0                  | 1                    |
+    And successful outcomes should equal 1
```
Comment (solomonahmed) [inline features/provider_bulk_load.feature:13]: Step definition for this : <successful/duplicate/problem/invalid/total> outcomes should equal <n> perhaps.

Maybe mark as an Issue for future refactoring?

File: features/step_definitions/dbconnect.rb [hunk 7]
```diff
@@ -0,0 +1,21 @@
+require 'ruby-oci8'
+
+
+module ChangeStartDate
```
Comment (solomonahmed) [inline features/step_definitions/dbconnect.rb:4]: Maybe create an Issue and cover evolving this at later date.

File: features/support/report_builder.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
+require 'report_builder'
```
Comment (solomonahmed) [inline features/support/report_builder.rb:1]: Ref the doc page for report_builder which covers these settings on Wiki or in README.md.

File: runner.sh [hunk 9]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (solomonahmed) [inline runner.sh:1]: Maybe another Issue is to create runner_stg or runner_uat variants if helpful.  And one for headless and normal possibly.

**Review by solomonahmed — APPROVED** [review]
LGTM.  Great work.  A number of points to carry over into Issues.  And some explanations needed to put into Readme's or on the Wiki.  These can be done at a future date.

File: Gemfile (hunk 1 above)
Comment (vimalsuresh) [reply Gemfile:17]: done

File: .circleci/config.yml (hunk 4 above)
Comment (vimalsuresh) [reply .circleci/config.yml:4]: created a wiki page. done

File: charts/cwa-tests/Chart.yaml (hunk 2 above)
Comment (vimalsuresh) [reply charts/cwa-tests/Chart.yaml:1]: done

File: charts/cwa-tests/Values.yaml (hunk 3 above)
Comment (vimalsuresh) [reply charts/cwa-tests/Values.yaml:1]: done

File: charts/cwa-tests/templates/pod.yml (hunk 5 above)
Comment (vimalsuresh) [reply charts/cwa-tests/templates/pod.yml:1]: done

File: runner.sh (hunk 9 above)
Comment (vimalsuresh) [reply runner.sh:1]: created an issue in git for above

File: features/support/report_builder.rb (hunk 8 above)
Comment (vimalsuresh) [reply features/support/report_builder.rb:1]: done

File: features/provider_bulk_load.feature (hunk 6 above)
Comment (vimalsuresh) [reply features/provider_bulk_load.feature:13]: created an issue 
https://github.com/ministryofjustice/laa-cwa/issues/312

File: features/step_definitions/dbconnect.rb (hunk 7 above)
Comment (solomonahmed) [reply features/step_definitions/dbconnect.rb]: As discussed this is covered by a Jira ticket.

**Review by solomonahmed — APPROVED** [review]
LGTM.  Great work!

**Pull request merged**